src/app/market/trades/trades.component.ts

```diff
@@ -0,0 +1,93 @@
+import _ from 'lodash';
+import moment from 'moment';
+import Promise from 'bluebird';
+import {Component} from '@angular/core';
+
+import {MarketDAO} from '../../shared/DAO/market.dao';
+import {ToastService} from '../../shared/services/toast.service';
+
+import t from '../../shared/defineTextToTranslate';
+
+@Component({
+  selector: 'app-market-trades',
+  templateUrl: 'trades.component.html'
+})
+export class TradesComponent {
+
+  public data: any;
+  public currencyFilter: string;
+  public interval: moment.unitOfTime.DurationConstructor;
+  public supportedPaymentsCurrencies: Array<Object>;
+  public supportedIntervals: Array<Object>;
+  public noData: boolean = false;
+  public loading: boolean = false;
+  public trades: Array<Object>;
+
+  constructor(private marketDAO: MarketDAO,
+              private toast: ToastService) {
+    this.supportedIntervals = [
+      {
+        label: t('MARKET.TRADE.INTERVAL.DAY'),
+        value: 'day'
+      },
+      {
+        label: t('MARKET.TRADE.INTERVAL.WEEK'),
+        value: 'week'
+      },
+      {
+        label: t('MARKET.TRADE.INTERVAL.MONTH'),
+        value: 'month'
+      },
+      {
+        label: t('MARKET.TRADE.INTERVAL.YEAR'),
+        value: 'year'
+      },
+    ];
+  }
+
+  ngOnInit() {
+    this.marketDAO.getAvailableMarkets().then((res: any) => {
+      this.supportedPaymentsCurrencies = _.chain(res).filter(o => o.lsymbol === 'BTC' && o.rtype === 'fiat').value();
```

Comment:
How about `supportedCurrencies`?

Reply:
supportedCurrencies is list of currencies pairs (BTC_USD, BTC_EUR etc.) available in market API to get data for this pair.

Reply:
But this is not marketApi. It's trades component.

Reply:
Trades in Market... historical list of trades.  Line 49: `this.marketDAO.getAvailableMarkets()`